edfi-repo-auditor/edfi_repo_auditor/auditor.py

```diff
@@ -59,98 +60,108 @@ def run_audit(config: Configuration) -> None:
+        CHECKLIST.ISSUES["description"]: get_message(CHECKLIST.DISCUSSIONS, not information["hasIssuesEnabled"]),
+        CHECKLIST.PROJECTS["description"]: get_message(CHECKLIST.PROJECTS, not information["hasProjectsEnabled"]),
+        CHECKLIST.DISCUSSIONS["description"]: get_message(CHECKLIST.DISCUSSIONS, information["discussions"]["totalCount"] == 0),
+        CHECKLIST.DELETES_HEAD["description"]: get_message(CHECKLIST.DELETES_HEAD, information["deleteBranchOnMerge"]),
+        CHECKLIST.USES_SQUASH["description"]: get_message(CHECKLIST.USES_SQUASH, information["squashMergeAllowed"]),
+        CHECKLIST.LICENSE_INFORMATION["description"]: get_message(CHECKLIST.LICENSE_INFORMATION, information["licenseInfo"] is not None),
+    } | dependabot_results
+
+
+def audit_alerts(client: GitHubClient, organization: str, repository: str, alerts: List[str]) -> dict:
+    vulnerabilities = [alerts for alerts in alerts
+                       if (alerts["createdAt"] < (datetime.now() - timedelta(ALERTS_WEEKS_SINCE_CREATED * 7)).isoformat() and
+                           alerts["securityVulnerability"]["advisory"]["severity"] in ALERTS_INCLUDED_SEVERITIES)]
+    total_vulnerabilities = len(vulnerabilities)
+
+    dependabot_enabled = client.has_dependabot_enabled(organization, repository)
+    return {
+        CHECKLIST.DEPENDABOT_ENABLED["description"]: CHECKLIST.DEPENDABOT_ENABLED["success"] if dependabot_enabled else CHECKLIST.DEPENDABOT_ENABLED["fail"],
+        CHECKLIST.DEPENDABOT_ALERTS["description"]: CHECKLIST.DEPENDABOT_ENABLED["success"] if total_vulnerabilities == 0 else CHECKLIST.DEPENDABOT_ALERTS["fail"],
     }
 
 
 def review_files(client: GitHubClient, organization: str, repository: str) -> dict:
-    files = {
-        CHECKLIST.README: False,
-        CHECKLIST.CONTRIBUTORS: False,
-        CHECKLIST.NOTICES: False,
-        CHECKLIST.LICENSE: False
-    }
+    file_audit: dict = {}
 
-    for file in files:
-        file_content = client.get_file_content(organization, repository, file)
-        if file_content:
-            files[file] = True
+    files_to_review = [CHECKLIST.README, CHECKLIST.CONTRIBUTORS, CHECKLIST.NOTICES, CHECKLIST.LICENSE]
 
-    return files
+    for file in files_to_review:
+        for filename in file["filename"]:
+            file_content = client.get_file_content(organization, repository, filename)
+            file_audit[file["description"]] = get_message(file, file_content is not None)
 
+    return file_audit
 
-def get_result(checklist: dict, rules: dict) -> int:
+
+def get_result(results: dict, rules: dict) -> int:
     score = 0
+
     for property in rules:
         try:
-            if (checklist[property] is True):
+            if (results[property] == DEFAULT_SUCCESS_MESSAGE):
                 score += rules[property]
         except KeyError:
             logger.error(f"Unable to read property {property} in results")
```

Comment:
Is this really an application error? I saw this come up a lot in the log. Either it seems like we have a coding error that needs to be fixed, or this is an expected situation and should be informational rather than an error.